pkg/resourceinterpreter/default/native/retain.go

```diff
@@ -122,3 +124,23 @@ func retainJobSelectorFields(desired, observed *unstructured.Unstructured) (*uns
 	}
 	return desired, nil
 }
+
+func retainK8sWorkloadReplicas(desired, observed *unstructured.Unstructured) (*unstructured.Unstructured, error) {
+	labels, exist, err := unstructured.NestedStringMap(desired.Object, "metadata", "labels")
+	if err != nil {
+		return nil, err
+	}
+
+	if _, hpaLabelExist := labels[util.ResourceScaledByHPALabel]; exist && hpaLabelExist {
+		replicas, exist, err := unstructured.NestedInt64(observed.Object, "spec", "replicas")
+		if err != nil || !exist {
+			return nil, fmt.Errorf("failed to get spec.replicas from deployment %s/%s", observed.GetName(), observed.GetNamespace())
```

Comment:
If this function could be used for other workloads, it might be better to use `observed.GetKind()` instead of `deployment`.

Or it might be better to change the name of function to make it specific.

Reply:
+1